chart/templates/configMap.yaml

```diff
@@ -5,9 +5,12 @@ metadata:
   labels: {{ include "rancher.labels" . | nindent 4 }}
     app.kubernetes.io/part-of: "rancher"
 data:
-  priorityClassName:  {{ .Values.priorityClassName }}
+  priorityClassName: {{ .Values.priorityClassName }}
   {{- if and .Values.webhook (kindIs "string" .Values.webhook) }}
-  rancher-webhook: {{  .Values.webhook | nindent 4 | squote }}
+  rancher-webhook: |
+    {{- range $key, $value := $.Values.webhook | fromYaml }}
+      {{ $key }}: {{ $value }}
+    {{- end }}
   {{- else if .Values.webhook }}
-  rancher-webhook: {{ toYaml .Values.webhook | nindent 4 | squote }}
+  rancher-webhook: {{ toYaml .Values.webhook | quote }}
   {{- end }}
```

Comment:
Since this is quite complicated, could you try to add some helm unit tests on the string stuff. We need to write down the various test cases involved here to future proof changes.

Reply:
Existing tests failed, which is good, so I adjusted them.

Reply:
Interesting, so it used to take a string input and then set a string output. Like in `'hostNetwork: true`. And now it will just use yaml. Ok.